src/app/layout.tsx

```diff
@@ -37,13 +41,43 @@ export default function RootLayout({
 }: Readonly<{
   children: React.ReactNode;
 }>) {
+  const localTheme = useMemo(() => {
+    if (typeof window === 'undefined') return dark;
```

Comment:
Pra não esquecermos desse fix, caso ocorra novamente, sugiro extrair a validação como um hook
```ts
const isVercelLoading = () => typeof window === 'undefined';
```

e salvar num arquivo de _hook_ próprio, fazendo o ajuste aqui:
```suggestion
    if (isVercelLoading()) return dark;
```